CRM/Utils/DeprecatedUtils.php

```diff
@@ -243,7 +243,18 @@ function _civicrm_api3_deprecated_formatted_param($params, &$values, $create = F
 
       case 'payment_instrument':
         require_once 'CRM/Core/PseudoConstant.php';
-        $values['payment_instrument_id'] = CRM_Core_PseudoConstant::getKey('CRM_Contribute_BAO_Contribution', 'payment_instrument_id', $value);
+        $key = CRM_Core_PseudoConstant::getKey('CRM_Contribute_BAO_Contribution', 'payment_instrument_id', $value);
+        if (!$key && ($key !== 0)) {
+          $pii_options = CRM_Contribute_BAO_Contribution::buildOptions('payment_instrument_id', 'create');
+          $lower_val = mb_strtolower($value);
```

Comment:
@agileware-fj is the strtolower biz just to try to ensure that case isn't an issue? Also just as a minor simplification how would this look

```php
$pii_options = CRM_Contribute_BAO_Contribution::buildOptions('payment_instrument_id', 'create');
foreach ($pii_options as $pik => $piv) {
  if (mb_strtolower($value) == mb_strtolower($piv)) {
    $key = $pik;
    break;
  }
}
```

Reply:
@seamuslee001 Yes, that's the "with case insensitivity" bit.  Actually I'm surprised I couldn't find an mb_strcasecmp, but there you have it.

Possibly not a major consideration but the idea with the assignment outside the loop was to prevent mb_strtolower from being called multiple times on the tested value (I don't trust PHP to figure this out for itself and multi-byte string operations are slow).

Reply:
@agileware-fj ah right ok makes sense then. There are some style issues with this PR at the moment see https://test.civicrm.org/job/CiviCRM-Core-PR/23509/checkstyleResult/new/

Reply:
@seamuslee001 just a FYI, we'll update the PR in the next few days

Reply:
@seamuslee001 We've updated the PR with style issues fix.